cpp/src/parquet/file_reader.cc

```diff
@@ -302,6 +303,17 @@ class SerializedFile : public ParquetFileReader::Contents {
 
   std::shared_ptr<FileMetaData> metadata() const override { return file_metadata_; }
 
+  std::shared_ptr<PageIndexReader> GetPageIndexReader() override {
+    if (!file_metadata_) {
+      throw ParquetException("Cannot get PageIndexReader as file metadata is not ready");
```

Comment:
nit: can we instruct the api caller what went wrong here (i.e. should they call a certain method?)

Reply:
Usually this won't happen if user calls one of the static Open() functions to create a ParquetFileReader instance. But if user calls the constructor directly and calls GetPageIndexReader() before Open() then this could happen. Add a comment to explain it and makes the exception message clear.